torch/csrc/jit/tensorexpr/ir_serializer.cpp

```diff
@@ -0,0 +1,557 @@
+      });
+}
+
+void torch::jit::tensorexpr::IRSerializer::visit(
+    torch::jit::tensorexpr::FloatImm const* v) {
+  JSON_CACHE_GUARD();
+  putJSON(
+      v,
+      {
+          {"expr_type", "FloatImm"},
+          {"value", v->value()},
+          {"dtype", serializeDtype(v->dtype())},
+      });
+}
+
+void torch::jit::tensorexpr::IRSerializer::visit(
+    torch::jit::tensorexpr::DoubleImm const* v) {
+  JSON_CACHE_GUARD();
+  putJSON(
+      v,
+      {
+          {"expr_type", "DoubleImm"},
+          {"value", v->value()},
+          {"dtype", serializeDtype(v->dtype())},
+      });
+}
+
+void torch::jit::tensorexpr::IRSerializer::visit(
+    torch::jit::tensorexpr::CharImm const* v) {
+  JSON_CACHE_GUARD();
+  putJSON(
+      v,
+      {
+          {"expr_type", "CharImm"},
+          {"value", v->value()},
+          {"dtype", serializeDtype(v->dtype())},
+      });
+}
+
+void torch::jit::tensorexpr::IRSerializer::visit(
+    torch::jit::tensorexpr::HalfImm const* v) {
+  JSON_CACHE_GUARD();
+  putJSON(
+      v,
+      {
+          {"expr_type", "HalfImm"},
+          {"value", static_cast<float>(v->value())},
+          {"dtype", serializeDtype(v->dtype())},
+      });
+}
+
+void torch::jit::tensorexpr::IRSerializer::visit(
+    torch::jit::tensorexpr::ByteImm const* v) {
+  JSON_CACHE_GUARD();
+  putJSON(
+      v,
+      {
+          {"expr_type", "ByteImm"},
+          {"value", v->value()},
+          {"dtype", serializeDtype(v->dtype())},
+      });
+}
+
+void torch::jit::tensorexpr::IRSerializer::visit(
+    torch::jit::tensorexpr::IntImm const* v) {
+  JSON_CACHE_GUARD();
+  putJSON(
+      v,
+      {
+          {"expr_type", "IntImm"},
+          {"value", v->value()},
+          {"dtype", serializeDtype(v->dtype())},
+      });
+}
+
+// TODO: change whether to include the parenthesis to the parent expression,
+// we need to look at the operator precedence to make the output simpler.
+template <typename Op>
+void visitBinaryOp(
+    const BinaryOpNode<Op>* v,
+    const std::string& op_str,
+    IRSerializer* serializer) {
+  if (serializer->cachedJSON(v)) {
+    return;
+  }
+  json value = {
+      {"lhs", serializer->visitAndSerialize(v->lhs())},
+      {"rhs", serializer->visitAndSerialize(v->rhs())},
+      {"dtype", serializeDtype(v->dtype())},
+      {"expr_type", op_str},
+  };
+  serializer->putJSON(v, value);
+}
+
+void IRSerializer::visit(const Add* v) {
+  visitBinaryOp(v, "Add", this);
+}
+
+void IRSerializer::visit(const Sub* v) {
+  visitBinaryOp(v, "Sub", this);
+}
+
+void IRSerializer::visit(const Mul* v) {
+  visitBinaryOp(v, "Mul", this);
+}
+
+void IRSerializer::visit(const Div* v) {
+  visitBinaryOp(v, "Div", this);
+}
+
+void IRSerializer::visit(const Mod* v) {
+  visitBinaryOp(v, "Mod", this);
+}
+
+void IRSerializer::visit(const Max* v) {
+  visitBinaryOp(v, "Max", this);
+}
+
+void IRSerializer::visit(const Min* v) {
+  visitBinaryOp(v, "Min", this);
+}
+
```

Comment:
You're discarding the `propagate_nans` field here.

The overlap in the IR between Min/Max and MinTerm/MaxTerm is confusing, it's not obvious which one should be used here. We should (separately) fix the comments on the Term nodes.